snp_motion_planning/src/planner_profiles.hpp

```diff
@@ -130,11 +131,31 @@ tesseract_planning::OMPLDefaultPlanProfile::Ptr createOMPLProfile(
   return profile;
 }
 
-std::shared_ptr<tesseract_planning::TrajOptPlanProfile> createTrajOptToolZFreePlanProfile()
+std::shared_ptr<tesseract_planning::TrajOptPlanProfile>
+createTrajOptToolZFreePlanProfile(const Eigen::VectorXd& cart_tolerance = Eigen::VectorXd::Zero(6))
 {
   auto profile = std::make_shared<tesseract_planning::TrajOptDefaultPlanProfile>();
-  profile->cartesian_coeff = Eigen::VectorXd::Constant(6, 1, 5.0);
-  profile->cartesian_coeff(5) = 0.0;
+  Eigen::VectorXd cart_coeff = Eigen::VectorXd::Constant(6, 1, 2.5);
+  cart_coeff(5) = 0.0;  // Set yaw to no cost
```

Comment:
If we're going to let the user specify tolerance, let's not set this cost to zero. This will let us support other planning cases as well

```suggestion
```

Reply:
Maybe we should just make this a parameter? If we set it to non-zero then it will try to drive back to the original orientation with the Cartesian cost coefficient.

Reply:
Yeah, I think parameterizing would be the best approach. I would be interested to see how the results differ when the cost is zero and non-zero. You would think the smoothing costs might force the z-orientation away from the nominal value despite the added cost, but I guess it would still have to compete with it

Reply:
I've implemented this now, I did a few tests with putting the coefficient to non-zero and if failed once and succeeded twice on the 3 toolpaths I tried.